### PR TITLE
Metrics

### PR DESCRIPTION
**Pull request by ericspaghetti**
Adds some basic metrics around:
* total sales
* total shoppers
* total inquiries sent
* total products listed

saves metrics to postgres db

Notes:
* when deploying to replit account, make sure Secrets has DATABASE_URL and OPENCAGE_API_KEY

### CODE REVIEW

File: pages/api/metrics/get-metrics.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  // Get everything after the `?`
+  const [, paramString] = uri.split('?');
+
+  // Return parameters
+  console.log(paramString)
+
+  return new URLSearchParams(paramString);
+};
+
+export default async function GetTotalSales(req: NextApiRequest, res: NextApiResponse) {
```
Comment (ericspaghetti) [inline pages/api/metrics/get-metrics.ts]:
```suggestion
export default async function GetMetrics(req: NextApiRequest, res: NextApiResponse) {
```

**Review by calvadev — APPROVED** [review]

File: pages/components/checkout-card.tsx [hunk 2]
```diff
@@ -142,6 +149,21 @@ export default function CheckoutCard({
     }
   }
 
+  const sendInvoicePaid = async (metricsInvoiceId: string) => {
```
Comment (thomasyeung687) [inline pages/components/checkout-card.tsx]: captureIncoicePaidMetric. I think that should be the naming convention for places where we are captureing data so we can locate all of them via shift+ctrl+f

File: pages/components/product-form.tsx [hunk 3]
```diff
@@ -133,6 +134,9 @@ export default function NewForm({
       tags.push(["t", category]);
     });
     let newListing = await PostListing(tags, passphrase);
+
+    PostListingMetric(newListing.id, tags);
```
Comment (thomasyeung687) [inline pages/components/product-form.tsx]: Same comment

File: pages/components/utility/nostr-helper-functions.ts [hunk 4]
```diff
@@ -337,6 +355,13 @@ async function generateNostrEventId(msg) {
   return hash;
 }
 
+export function getPubKey() {
```
Comment (thomasyeung687) [inline pages/components/utility/nostr-helper-functions.ts:359]: This should just be the decryptednpub from getlocalstorageData I beleive

**Review by thomasyeung687 — APPROVED** [review]
LGTM

**Pull request merged**